engine/engine.go

```diff
@@ -250,43 +257,69 @@ func (e *Engine) PublishLatest(ctx context.Context) (cid.Cid, error) {
 	return adCid, nil
 }
 
-// PublishLatestHTTP publishes the latest existing advertisement to a specific
-// indexer.
-func (e *Engine) PublishLatestHTTP(ctx context.Context, indexerHost string) (cid.Cid, error) {
+// PublishLatestHTTP publishes the latest existing advertisement to the
+// specific indexers.
+func (e *Engine) PublishLatestHTTP(ctx context.Context, announceURLs ...*url.URL) (cid.Cid, error) {
 	adCid, err := e.latestAdToPublish(ctx)
 	if err != nil {
 		return cid.Undef, err
 	}
 
+	err = e.httpAnnounce(ctx, adCid, announceURLs)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return adCid, nil
+}
+
+func (e *Engine) httpAnnounce(ctx context.Context, adCid cid.Cid, announceURLs []*url.URL) error {
 	ai := &peer.AddrInfo{
 		ID: e.h.ID(),
 	}
 
 	switch e.pubKind {
 	case NoPublisher:
 		log.Info("Remote announcements disabled")
-		return cid.Undef, nil
+		return nil
 	case DataTransferPublisher:
 		ai.Addrs = e.h.Addrs()
 	case HttpPublisher:
 		maddr, err := hostToMultiaddr(e.pubHttpListenAddr)
 		if err != nil {
-			return cid.Undef, err
+			return err
 		}
 		proto, _ := multiaddr.NewMultiaddr("/http")
 		ai.Addrs = append(ai.Addrs, multiaddr.Join(maddr, proto))
 	}
 
-	cl, err := httpclient.New(indexerHost)
-	if err != nil {
-		return cid.Undef, err
+	errChan := make(chan error)
+	for _, u := range announceURLs {
+		// Send HTTP announce to indexers concurrently.
```

Comment:
Do we need to close `errChan` somewhere?

Reply:
No, all announce goroutines put a response onto `errChan`, and the look that collects the response gets a response for each goroutine, without needing to wait for the channel to close.  Once a response for each announce goroutine has been received, then all goroutines have completed.